packages/agents-manager/src/components/agent-dock/agent.tsx

```diff
@@ -0,0 +1,220 @@
+import { AgentsManagerSelect } from '@automattic/data-stores';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useCallback, useEffect, useRef } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { useNavigate, useParams } from 'react-router-dom';
+import { API_BASE_URL } from '../../constants';
+import useLoadConversation from '../../hooks/use-load-conversation';
+import { AGENTS_MANAGER_STORE } from '../../stores';
+import { lastConversationCache } from '../../utils/conversation-cache';
+import BigSkyIcon from '../big-sky-icon';
+import ChatHeader, { type Options as ChatHeaderOptions } from '../chat-header';
+import ChatMessageSkeleton from '../chat-message-skeleton';
+import { AI } from '../icons';
+import type { ComponentType } from 'react';
+
+interface AgentProps {
+	agentConfig: UseAgentChatConfig;
+	emptyViewSuggestions: Suggestion[];
+	getChatHeaderOptions: () => ChatHeaderOptions;
+	isDocked: boolean;
+	messageRenderer: ComponentType< { children: string } >;
+	onClose: () => void;
+}
+
+export default function Agent( {
+	agentConfig,
+	emptyViewSuggestions,
+	getChatHeaderOptions,
+	isDocked,
+	messageRenderer,
+	onClose,
+}: AgentProps ) {
+	const { chatId = '' } = useParams< { chatId?: string } >();
+	const { setIsOpen, setSessionId } = useDispatch( AGENTS_MANAGER_STORE );
+	const { isOpen } = useSelect( ( select ) => {
+		const store: AgentsManagerSelect = select( AGENTS_MANAGER_STORE );
+		return store.getAgentsManagerState();
+	}, [] );
+	const navigate = useNavigate();
+
+	const isLoadingRef = useRef( false );
+	const loadedSessionIdRef = useRef< string | null >( null );
+
+	const agentId = agentConfig.agentId;
+
+	const { messages, suggestions, isProcessing, error, loadMessages, onSubmit } =
+		useAgentChat( agentConfig );
+
+	// Memoized callback for when conversation loads from server
+	const onLoaded = useCallback(
+		async ( loadedMessages: Message[], serverSessionId: string ) => {
+			const agentManager = getAgentManager();
+			const agentKey = agentId;
+
+			// Agent should already be created by useAgentChat, but check just in case
+			if ( ! agentManager.hasAgent( agentKey ) ) {
+				await agentManager.createAgent( agentKey, {
+					...agentConfig,
+					sessionId: serverSessionId,
+				} );
+			}
+
+			// Use loadMessages instead of direct replaceMessages to ensure React state updates
+			await loadMessages( loadedMessages );
+
+			// Update the agent's sessionId so future messages use the correct session
+			agentManager.updateSessionId( agentKey, serverSessionId );
+
+			// Only update session if it changed (prevents unnecessary re-renders)
+			if ( chatId !== serverSessionId ) {
+				try {
+					if ( ! serverSessionId ) {
+						// eslint-disable-next-line no-console
+						console.warn( '[AgentDock] Attempted to apply empty session ID' );
+						return;
+					}
+
+					navigate( `/chat/${ serverSessionId }`, { replace: true } );
```

Comment:
Double-check this navigation:
- Determine whether we actually need it
- If we do, place it in an appropriate location
- Make sure there are no redundant API calls